src/Importer.php

```diff
@@ -320,47 +323,43 @@ private function log($message)
         $this->logger->debug($message);
     }
 
+    /**
+     * Setting the XML from $xmlString or a $xmlFile
+     */
     private function setXml()
     {
-        // Enable user error handling while validating input
-        libxml_clear_errors();
-        $useInternalErrors = libxml_use_internal_errors(true);
-
         $this->log("Load XML ...");
-        $xml = null;
 
-        if ($this->xmlFile !== null) {
-            $xml = new DOMDocument();
-            $xml->load($this->xmlFile);
-            if (! $xml) {
-                throw new MetadataImportInvalidXmlException('XML is not well-formed.');
-            }
-        } else {
-            $xml = new DOMDocument();
-            $xml->loadXML($this->xmlString);
-            if (! $xml) {
-                throw new MetadataImportInvalidXmlException('XML is not well-formed.');
+        try {
+            if ($this->xmlFile !== null) {
```

Comment:
Ich denke es wäre besser, wie in der `DOMDocument` Klasse unterschiedliche Funktionen für Dateien und XML-Strings zu verwenden.